File: python/2731.py
from typing import List

class Solution:
    def sumDistance(self, nums: List[int], s: str, d: int) -> int:
        MOD = 10 ** 9 + 7
        for i, ch in enumerate(s):
            nums[i] += d if ch == 'R' else -d
        nums.sort()
        tmp = 0
        ans = 0
        for i, num in enumerate(nums):
            ans += i * num - tmp
            tmp += num
        return ans % MOD

File: go/2731.go
package main

import "sort"

func sumDistance(nums []int, s string, d int) int {
	const MOD = 1_000_000_007
	for i, ch := range s {
		nums[i] += d * int(ch&2-1)
	}
	sort.Ints(nums)
	tmp, ans := 0, 0
	for i, num := range nums {
		ans = (ans + i*num - tmp) % MOD
		tmp += num
	}
	return ans
}
